### PR TITLE
Feature/overview of plants

### PR DESCRIPTION
**Pull request by evabergold**
PR for [UserStory Overview of Plants](https://github.com/users/SabinePfanner/projects/2/views/1?pane=issue&itemId=64300574)

**Please test the preview only in Mobile view, not optimized for Desktop yet!**

Sample object from MongoDB to see the data structure: 
```
{"_id":{"$oid":"665446547308e87f424ec692"},"name":"Tomato","botanicalName":"Solanum lycopersicum","waterDemand":"High","nutrientDemand":"High","growingConditions":"Sunny","placement":"Bed/Pot","perennial":false,"frostSensitive":true,"cropType":"Vegetable","image":"/images/tomato.jpg"}


```

Bildschirmfoto 2024-05-28 um 11 12 39

### CODE REVIEW

File: components/cardList.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: 1fr;
+  grid-gap: 1rem;
+  grid-auto-flow: row;
+`;
+
+export default function CardList() {
+  const { data: plants, error, isLoading } = useSWR("/api/plants");
+  console.log("plants in cardList: ", plants);
```
Comment (doemser) [inline components/cardList.js]: always remove console.logs!

File: components/cardList.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (!plants) {
+    return;
+  }
+
+  return (
+    <>
```
Comment (doemser) [inline components/cardList.js]: remove fragments

File: components/cardList.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      <StyledList>
+        {plants.map((plant) => {
+          console.log(plant);
```
Comment (doemser) [inline components/cardList.js]: remove console.logs

File: components/image.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import Image from "next/image.js";
+import styled from "styled-components";
+
+export const StyledImage = styled(Image)`
+  width: 80%;
```
Comment (doemser) [inline components/image.js]: might "squansh" your images

**Review by doemser — CHANGES_REQUESTED** [review]
Good first one

File: components/card.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { StyledImage } from "./image.js";
+import styled from "styled-components";
+
+const Article = styled.article`
+  //padding: 0.5rem;
```
Comment (doemser) [inline components/card.js]: remove styled component if it is not doing anything

File: components/card.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+`;
+
+const Figure = styled.figure`
+  position: relative;
```
Comment (doemser) [inline components/card.js]: should be default

File: components/image.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import Image from "next/image.js";
+import styled from "styled-components";
+
+export const StyledImage = styled(Image)``;
```
Comment (doemser) [inline components/image.js]: no need for styled components if no styling is applied

**Review by doemser — CHANGES_REQUESTED** [review]

**Review by FelixNeumayr — APPROVED** [review]
approved

**Pull request merged**